miprometheus/workers/offline_trainer.py

```diff
@@ -265,18 +267,19 @@ def run_experiment(self):
                         # Perform validation.
                         self.validate_on_batch(self.validation_batch, episode, epoch)
 
+                        # Aggregate statistics, but do not display them in log.
+                        self.aggregate_and_export_statistics(self.model, self.validation_problem, 
+                                self.validation_stat_col, self.validation_stat_agg, episode, '[Partial Validation]', False)
+
                         # Save the model using the latest validation statistics.
-                        self.model.save(self.model_dir, self.validation_stat_col)
+                        self.model.save(self.model_dir, training_status, self.validation_stat_agg)
 
                     # III. The episodes number limit has been reached.
                     if episode+1 >= self.episode_limit:
-                        termination_cause = "Episode Limit reached"
-                        break
+                        training_status = "Episode Limit reached"
```

Comment:
So here you are checking `if episode+1 >= self.episode_limit`, but you changed the initial value of `episode`, and where it is incremented in the loop. Does that not cause the training to finish one episode earlier?

Reply:
Double checked it in both workers, it is ok.
For example, setting episode_limit = 133 results in collection of statistics up to episode 132, aggregation also happens in episode 132.